src/Web/Program.cs

```diff
@@ -21,7 +23,19 @@
 
 builder.Logging.AddConsole();
 
-Microsoft.eShopWeb.Infrastructure.Dependencies.ConfigureServices(builder.Configuration, builder.Services);
+// Configure SQL Server
+var credential = new ChainedTokenCredential(new AzureDeveloperCliCredential(), new DefaultAzureCredential());
+builder.Configuration.AddAzureKeyVault(new Uri(builder.Configuration["AZURE_KEY_VAULT_ENDPOINT"]), credential);
+builder.Services.AddDbContext<CatalogContext>(c =>
+{
+    var connectionString = builder.Configuration[builder.Configuration["AZURE_SQL_CATALOG_CONNECTION_STRING_KEY"]];
```

Comment:
Do we need 2 builder.Configuration statements?

How do we go from getting the key to getting the connection string from key vault?

Reply:
The internal builder.Configuration will get the connection string key, and the `AddAzureKeyVault()` method will adds an `Microsoft.Extentions.Configuration.IConfigurationProvider` that reads configuration values from the Azure KeyVault. Therefore, the external builder.Configuration can get the corresponding secret value according to the key.

In addition, we refer to the Azure-Samples/todo-csharp-sql template, which is also used in this way. We have also tested it and it works fine. If you think this is inappropriate, we can change to the previous method and add a new method similar to `getSqlConnStr` to get the secret value from keyVault.

Reply:
Okay, I see what it is doing now.  Thanks,